### PR TITLE
added config for wishlist sidebar in config

### PR DESCRIPTION
**Pull request by torhoehn**
### Description

Until now there wasn't a config field to hide wishlist from sidebar.

### Manual testing scenarios

1. Open Backend
2. Go to Stores -> Configuration -> Customers -> Wish List
3. Set 'Show in Sidebar' to 'No'
4. Wishlist will be hidden from sidebar

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19554) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=19554) before we can accept your contribution.<br/><hr/>**Torben Höhn** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by magento-engcom-team** [comment]
Hi @torhoehn. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by torhoehn** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @torhoehn. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @torhoehn, here is your [new Magento instance](https://pr-19554.instances.magento-community.engineering).
Admin access: [https://pr-19554.instances.magento-community.engineering/admin](https://pr-19554.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by Karlasa** [comment]
@torhoehn Great idea 👍 
Maybe update en_US.csv file also? :)

**Comment by torhoehn** [comment]
> @torhoehn Great idea 👍
> Maybe update en_US.csv file also? :)

@Karlasa 
Of course, updated it!

**Review by Karlasa — APPROVED** [review]

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3617](https://jira.corp.magento.com/browse/ENGCOM-3617) has been created to process this Pull Request

File: app/code/Magento/Wishlist/etc/adminhtml/system.xml [hunk 1]
```diff
@@ -39,6 +39,10 @@
                     <label>Enabled</label>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                 </field>
+                <field id="show_in_sidebar" translate="label" type="select" sortOrder="5" showInDefault="1" showinWebsite="1" showInStore="1" canRestore="1">
```
Comment (sivaschenko) [inline app/code/Magento/Wishlist/etc/adminhtml/system.xml]:
```suggestion
                <field id="show_in_sidebar" translate="label" type="select" sortOrder="5" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
```
This typo results into exception on Configuration page

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@torhoehn please see my code update suggestion. Currently there is an exception on Stores Configuration page as `system.xml` does not pass the validation

**Comment by magento-engcom-team** [comment]
Hi @torhoehn. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**